### PR TITLE
[Chat][Bug][Feature]Add Peft support and fix the bug when ptx_coef is enabled

### PR DESCRIPTION
**Pull request by yynil**
[Chat][Bug][Feature]Add Peft support and fix the bug when ptx_coef is enabled

## 📌 Checklist before creating the PR

- [ x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number
[issues/3432](https://github.com/hpcaitech/ColossalAI/issues/3432)
[issues/3454](https://github.com/hpcaitech/ColossalAI/issues/3454)

## 📝 What does this PR do?
1. Fix bug of wrong data fetching when pretrained data loss is enabled.
2. Add Huggingface Peft model support for SFT and Prompts(aka. Stage-1 and Stage-3) training.

## 💥 Checklist before requesting a review

- [ x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ x] I have performed a self-review of my code
- [ x] I have added thorough tests.
- [ x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/Chat/examples/test_pretrained.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+提问：失信被执行人能坐火车软卧吗 回答：失信被执行人可以坐火车硬卧。失信被执行人不能乘坐飞机、轮船二等以上舱位，不能乘坐G字头高铁所有席别、D字头动车一等座及以上席别、普通列车软卧及以上席别，其它动车组列车一等以上座位等其它非生活和工作必需的消费行为是禁止的。但乘坐其它席别的火车仍不会受到影响。限制消费措施一般由申请执行人提出书面申请，经人民法院审查决定；必要时人民法院可以依职权决定。人民法院决定采取限制消费措施的，应当向被执行人发出限制消费令。限制消费令由人民法院院长签发。限制消费令应当载明限制消费的期间、项目、法律后果等内容。人民法院决定采取限制消费措施的，可以根据案件需要和被执行人的情况向有义务协助调查、执行的单位送达协助执行通知书，也可以在相关媒体上进行公告。法律依据：《最高人民法院关于限制被执行人高消费及有关消费的若干规定》第九条在限制消费期间，被执行人提供确实有效的担保或者经申请执行人同意的，人民法院可以解除限制消费令；被执行人履行完毕生效法律文书确定的义务的，人民法院应当在本规定第六条通知或者公告的范围内及时以通知或者公告解除限制消费令。
```
Comment (ht-zhou) [inline applications/Chat/examples/test_pretrained.txt:1]: Can you delete the test txt

File: applications/Chat/coati/trainer/ppo.py [hunk 2]
```diff
@@ -92,9 +92,10 @@ def training_step(self, experience: Experience) -> Dict[str, float]:
 
         # ptx loss
         if self.ptx_coef != 0:
-            ptx = next(iter(self.pretrain_dataloader))['input_ids'].to(torch.cuda.current_device())
```
Comment (ht-zhou) [inline applications/Chat/coati/trainer/ppo.py:95]: It's not a bug here but change the form right?

File: applications/Chat/examples/test_pretrained.txt (hunk 1 above)
Comment (yynil) [reply applications/Chat/examples/test_pretrained.txt:1]: no problem

Comment (yynil) [reply applications/Chat/examples/test_pretrained.txt]: > 

Done

**Comment by ht-zhou** [comment]
Thank you for your nice work . 
We suggest you to make a folder under example folder named "community" to store your code there.
Or you can choose to add a flag named "peft".
Also, we suggest you to delete the Chinese comment.

**Comment by yynil** [comment]
> Thank you for your nice work . We suggest you to make a folder under example folder named "community" to store your code there. Or you can choose to add a flag named "peft". Also, we suggest you to delete the Chinese comment.

Very nice suggestions. I just move all my codes to community folder and remove the Chinese comments while keeping the original files untackled.

**Review by ht-zhou — APPROVED** [review]

**Pull request merged**